### PR TITLE
Auto-Restore UI normalization

### PR DESCRIPTION
**Pull request by alpaix**
Resolves NuGet/Home#3633, NuGet/Home#3789, NuGet/Home#3835,
NuGet/Home#3884.

The purpose of this change is to make consistent UX in all restore
scenarios: auto-restore, on-build restore, and solution restore. At the
same time each scenario has some nuances, whereas:
- Auto-restore and solution restore should write log messages in PM console
output window.
- On-build restore writes inline log messages in build output window.

Introduced new console type: `BuildOutputConsole`. It allows writing into
the VS build output window via `IOutputConsole` abstraction API.

Moved `SolutionRestoreCommand` into the solutuion restore package. As the
latter loads on solution open so the restore command should be available
at this point. At the same time the command invokation shouldn't trigger
loading of the legacy package.

Also had to do some refactoring in order to consume `VsConsole` API in
Solution Restore implementation classes:
- Moved all `SolutionRestore*` artifacts into
  `NuGet.SolutionRestoreManager` project.

Eliminated dead code including remnants of debug console and standalone
console.

//cc @emgarten @jainaashish @mishra14 @rohit21agrawal @rrelyea

### CODE REVIEW

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/NuGet.SolutionRestoreManager.csproj [hunk 1]
```diff
@@ -22,20 +22,36 @@
     <Reference Include="envdte, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
       <EmbedInteropTypes>False</EmbedInteropTypes>
     </Reference>
+    <Reference Include="envdte80, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager/NuGet.SolutionRestoreManager.csproj]: Is this needed? envdte 8.0.0.0 is referenced above it? I think these names may be dupes but I'm unsure.

**Comment by jainaashish** [comment]
> Moved SolutionRestoreCommand into the solutuion restore package. As the
> latter loads on solution open so the restore command should be available
> at this point. At the same time the command invokation shouldn't trigger
> loading of the legacy package.
> 
> Also had to do some refactoring in order to consume VsConsole API in
> Solution Restore implementation classes:
> 
> Moved all SolutionRestore* artifacts into
> NuGet.SolutionRestoreManager project.

before starting with the code review, I would advice to see if this part had any perf regression with current solution load time since we're already dealing with a perf regression in same area and this change may make it worse.

**Review by emgarten — APPROVED** [review]
Glad to see these scenarios all working in the same way!

**Comment by alpaix** [comment]
@jainaashish Will arrange a perf test run to verify whether this change increase startup time.

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/NuGet.SolutionRestoreManager.csproj (hunk 1 above)
Comment (alpaix) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager/NuGet.SolutionRestoreManager.csproj]: `EnvDTE80.DTE2` is needed in certain cases to get access to advanced extensibility functionality.

**Review by jainaashish — CHANGES_REQUESTED** [review]
waiting for the perf test result

**Comment by alpaix** [comment]
Performance test ValBuild Dev15 $/DevDiv/auxsrc/vs/d15rel.26013.00.alpanov.161213.160152 RPS V3 DDRIT TH2 did not discover any regressed scenarios.

**Review by jainaashish — APPROVED** [review]

**Pull request merged**